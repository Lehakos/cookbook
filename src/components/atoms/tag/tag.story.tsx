import React from 'react';
import { storiesOf } from '@storybook/react';

import { Tag } from './tag';

const stories = storiesOf('Atoms|Tag', module);

stories
  .add('Default', () => <Tag>Картофель</Tag>)
  .add('Dark background', () => <Tag color="#38618C">Картофель</Tag>)
  .add('Light background', () => <Tag color="#F9ADA0">Картофель</Tag>);
