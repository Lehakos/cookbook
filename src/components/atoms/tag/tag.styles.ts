import styled from 'styled-components';
import { prop } from 'styled-tools';
import { readableColor } from 'polished';

type WrapperProps = { bgColor: string };

export const Wrapper = styled.div.attrs<WrapperProps>(({ bgColor }) => ({
  style: {
    backgroundColor: bgColor,
  },
}))<WrapperProps>`
  display: inline-block;
  padding: ${prop('theme.space.2')}px ${prop('theme.space.3')}px;
  border: none;

  text-align: center;
  font-size: ${prop('theme.fontSizes.0')};
  color: ${({ bgColor, theme }) =>
    readableColor(bgColor, theme.colors.text, '#fff')};

  border-radius: ${prop('theme.radii.0')}px;
`;
