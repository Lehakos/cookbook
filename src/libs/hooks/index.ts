export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
