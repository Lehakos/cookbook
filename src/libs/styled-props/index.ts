export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
