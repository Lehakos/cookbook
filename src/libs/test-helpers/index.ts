export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
