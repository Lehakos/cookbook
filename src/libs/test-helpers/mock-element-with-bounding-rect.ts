export const mockElementWithBoundingRect = ({
  width = 100,
  height = 100,
  top = 0,
  left = 0,
}) => {
  const right = left + width;
  const bottom = top + height;

  Object.defineProperties(Element.prototype, {
    offsetLeft: {
      get() {
        return left;
      },
      configurable: true,
    },
    offsetTop: {
      get() {
        return top;
      },
      configurable: true,
    },
    offsetHeight: {
      get() {
        return height;
      },
      configurable: true,
    },
    offsetWidth: {
      get() {
        return width;
      },
      configurable: true,
    },
  });

  Element.prototype.getBoundingClientRect = jest.fn(() => {
    return {
      width,
      height,
      top,
      left,
      bottom,
      right,
    };
  });

  const element = document.createElement('div');

  return element;
};
