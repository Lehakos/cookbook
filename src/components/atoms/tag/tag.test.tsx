import React from 'react';

import { render } from 'libs/test-helpers';

import { Tag } from './tag';

describe('<Tag />', () => {
  it('renders properly', () => {
    const { container } = render(<Tag />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
