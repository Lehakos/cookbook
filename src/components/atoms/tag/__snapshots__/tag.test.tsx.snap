// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> renders properly 1`] = `
.c0 {
  display: inline-block;
  padding: 4px 8px;
  border: none;
  text-align: center;
  font-size: 0.75rem;
  color: #3D3E3C;
  border-radius: 2px;
}

<div
  class="c0"
  style="background-color: rgb(226, 226, 226);"
/>
`;
